soc/nordic/common/dmm.h

```diff
@@ -33,8 +33,10 @@ extern "C" {
 /* Determine required alignment of the data buffers in specified memory region.
  * Cache line alignment is required if region is cacheable and data cache is enabled.
  */
-#define DMM_REG_ALIGN_SIZE(node_id) \
-	(DMM_IS_REG_CACHEABLE(node_id) ? CONFIG_DCACHE_LINE_SIZE : sizeof(uint8_t))
+#define DMM_REG_ALIGN_SIZE(node_id)							\
+	COND_CODE_1(CONFIG_DCACHE,							\
+	   (DMM_IS_REG_CACHEABLE(node_id) ? CONFIG_DCACHE_LINE_SIZE : sizeof(uint8_t)),	\
```

Comment:
Missing one pair of parentheses. The outermost ones will be taken out by the `COND_CODE_1()` macro, so the result will be different than before and may potentially break some code.
```suggestion
	   ((DMM_IS_REG_CACHEABLE(node_id) ? CONFIG_DCACHE_LINE_SIZE : sizeof(uint8_t))),	\
```
But perhaps it will be cleaner and easier to handle `CONFIG_DCACHE` outside of the macro definition. Especially that `DMM_IS_REG_CACHEABLE()` needs to check for it as well. How about:
```
#ifdef CONFIG_DCACHE

/* Determine if memory region is cacheable. */
#define DMM_IS_REG_CACHEABLE(node_id)					     \
	COND_CODE_1(DT_NODE_HAS_PROP(node_id, zephyr_memory_attr),	     \
		((DT_PROP(node_id, zephyr_memory_attr) & DT_MEM_CACHEABLE)), \
		(0))

/* Determine required alignment of the data buffers in specified memory region.
 * Cache line alignment is required if region is cacheable and data cache is enabled.
 */
#define DMM_REG_ALIGN_SIZE(node_id) \
	(DMM_IS_REG_CACHEABLE(node_id) ? CONFIG_DCACHE_LINE_SIZE : sizeof(uint8_t))

#else
#define DMM_IS_REG_CACHEABLE(node_id) 0
#define DMM_REG_ALIGN_SIZE(node_id) sizeof(uint8_t)
#endif /* CONFIG_DCACHE */
```

Reply:
Fair enough. Applied the long version.